src/printer.js

```diff
@@ -1955,6 +1955,26 @@ function shouldGroupLastArg(args) {
     (!penultimateArg || penultimateArg.type !== lastArg.type);
 }
 
+function shouldGroupFirstArg(args) {
+  if (args.length !== 2) {
+    return false;
+  }
+
+  const firstArg = args[0];
+  const secondArg = args[1];
+  return (firstArg.type === 'FunctionExpression' ||
+    (firstArg.type === 'ArrowFunctionExpression' &&
+      firstArg.body.type === 'BlockStatement')) &&
+      // Only allow one simple trailing arg.
```

Comment:
I'm trying to keep to James' request in #889 that it be "primitive" (literal or identifier).

I think it should allow MemberExpressions that only contain identifiers e.g. `this.prop.timeout`.  

The binary expression `timeout * 1000` should be disallowed because it's 2 things not one. (This doesn't look too bad split onto a seperate line, to my eye. Subjective!)

CallExpressions are interesting. `this.props.getTimeout()` doesn't look to bad, but if we are choosing between `this.timeout`, `this.getTimeout()` and `this.getTimeout(args)`, I think it's less surprising to exclude all CallExpressions than exclude CallExpressions with args.